enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,157 @@
+Relying soley on automated API detection to block upgrades could result in blocking for APIs that are not actively used any longer or not blocking because an application was not active during the check.
+Having a human in the loop to drive the analysis and make the final decision is the surest way to prevent customers from upgrading from 4.8.z to 4.9.z and encountering missing API related breakage.
+
+### Goals
+
+* Acknowledgment requires the explicit update of a cluster resource by an administrator.
+* For the initial release in 4.8.z, administrator acknowledgment of API removal will be required before a minor level upgrade to 4.9 is allowed.
+* Upgrade blocking mechanism must be generic such that it can be repurposed in the future.
+* It must be clear what an administrator is acknowledging.
+* Administrators can bypass the upgrade blocking mechanism by using the existing CVO "force" implementation.
+* The UI will clearly indicate when this upgrade blocking mechanism is blocking upgrades, why, and what needs to be done to clear the block.
+* This upgrade blocking mechanism must be compatible with SD, ACM, IKS/ROKS, Hive, and Hypershift.
+
+### Non-Goals
+
+## Proposal
+
+### General Implementation
+
+CVO's existing framework for blocking minor level upgrades will be leveraged.
+The existing framework periodically checks upgradeable conditions.
+A new [upgradeable check method][upgradeableCheck-interface] will be added to the existing framework to implement this upgrade blocking mechanism.
+Use of this existing framework also allows this new upgrade blocking mechanism to be bypassed by using the `force` upgrade option.
+
+The new method will check a newly defined configmap for one or more predefined keys.
+Each key represents a specific upgrade blocking condition that must be acknowledged before minor level upgrades are allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the predefined keys in the configmap with a value of `true`.
+For each key blocking an upgrade, CVO will generate a detailed message pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
```

Comment:
How the admin will be communicated or get to know that the config needs to me modified. I guess from the clusterversion status condition? or an alert? Also how we will we communicate the admin the exact command to edit the config map e.g. the patch command or the `oc edit command` .

Reply:
The way it's written, since I saw no requirement about an alert, is the admin will be notified exactly as an admin is now when CVO blocks a minor release upgrade for other reasons. As far as what the admin needs to do before to get pass the block, the Condition message itself is to be very detailed and point admin to updated doc's with details:

> For each key blocking an upgrade, CVO will generate a detailed message pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.`

We could add an alert if that's what the consensus is but then that makes this minor level upgrade blocker differ from the others. Does it need to be?

Reply:
I think as long as we are satisfied w/ the existing UX when an operator is in an "upgradeable=false" status, that is fine.